### PR TITLE
csi: provide proper Expansion option to External volume

### PR DESCRIPTION
**Pull request by amarts**
Updates: #596 
Signed-off-by: Amar Tumballi <amar@kadalu.io>

### CODE REVIEW

**Comment by amarts** [comment]
Haven't yet tested. Need some help in testing this.

File: csi/controllerserver.py [hunk 1]
```diff
@@ -590,7 +596,42 @@ def ControllerExpandVolume(self, request, context):
...
+                "gluster",
+                "volume",
+                "quota",
+                "%s" % existing_volume.gvolname,
+                "limit-usage",
+                "/%s" % existing_volume.volpath,
+                "%s" % expansion_requested_pvsize * 0.05,
+            ]
+            try:
+                execute(*quota_cmd)
```
Comment (aravindavk) [inline csi/controllerserver.py]: Repeat code can be moved to a function

File: csi/controllerserver.py [hunk 2]
```diff
@@ -590,7 +596,42 @@ def ControllerExpandVolume(self, request, context):
...
+            except CommandException as err:
+                errmsg = "Unable to set Gluster Quota via ssh"
+                logging.error(logf(errmsg, error=err))
+                context.set_details(errmsg)
+                context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+                return csi_pb2.CreateVolumeResponse()
```
Comment (aravindavk) [inline csi/controllerserver.py]: return csi_pb2.ControllerExpandVolumeResponse()

File: csi/controllerserver.py [hunk 3]
```diff
@@ -545,6 +545,11 @@ def ControllerExpandVolume(self, request, context):
 
         # Get existing volume
         existing_volume = search_volume(request.volume_id)
+        if existing_volume.extra['kformat'] == 'non-native':
```
Comment (leelavg) [inline csi/controllerserver.py:562]: How about setting `allowVolumeExpansion` dynamically [here](https://github.com/kadalu/kadalu/blob/devel/templates/storageclass-kadalu.custom.yaml.j2#L8) and [here](https://github.com/kadalu/kadalu/blob/devel/templates/external-storageclass.yaml.j2#L7) while deploying `sc` itself?

Then we might get a good side effect of getting this field reflected in `kubectl get pvc` and k8s will not allow expansion as well. Feasible?

File: csi/controllerserver.py (hunk 2 above)
Comment (leelavg) [reply csi/controllerserver.py]: ig, it's better to have an `else` as well, like some users may not use gluster quota and in that case exapandvol should just log a msg?

**Review by aravindavk — APPROVED** [review]

File: csi/controllerserver.py (hunk 3 above)
Comment (amarts) [reply csi/controllerserver.py:562]: allowVolumeExpansion is not present? regardless, should be a separate PR IMO.

Comment (amarts) [reply csi/controllerserver.py:562]: After checking, yes, it should be added with this PR itself. Done now. Thanks.

**Comment by amarts** [comment]
@yugyama if you get chance, can you also have a look on this PR? Any concerns?

**Comment by yugyama** [comment]
> @yugyama if you get chance, can you also have a look on this PR? Any concerns?

Looks good to me

**Pull request merged**